ipaserver/plugins/sudorule.py

```diff
@@ -573,6 +573,32 @@ class sudorule_remove_deny_command(LDAPRemoveMember):
     member_count_out = ('%i object removed.', '%i objects removed.')
 
 
+def pre_callback_process_external_objects(memberattr,
+                                          ldap, dn, found, not_found,
+                                          *keys, **options):
+    for o_type in ('user', 'group'):
+        dn = add_external_pre_callback(o_type, ldap, dn, keys, options)
+        if 'trusted_objects' in options:
```

Comment:
A little explanation may help here. filtered_objects contains the found users/groups that are not trusted objects? In this case the algo looks wrong to me as it won't work if multiple trusted objects are added in the same command.

Reply:
Actually, multiple objects do work but I found a bug in cooperation between pre-callback and actual member entry updates in `LDAPAddMember` which leads to duplication of added external users. Working on it.

Reply:
Fixed